src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,266 @@
+  MAX_RING_PARTICLES,
+  EASING,
+  RING_RADIUS,
+  RING_COVERAGE,
+  EXPLODE_RANGE,
+  EXPLODE_OFFSET,
+  FIREWORK_LIFESPAN,
+  LIFESPAN_DECREMENT,
+  VELOCITY_CHANGE,
+  TRAIL_SIZE,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+  const fireworks: Firework[] = [];
+  const ringParticles: RingParticle[] = [];
+  let gravity: p5.Vector;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+    gravity = p.createVector(0, 0.095);
+    p.stroke(255);
+    p.strokeWeight(4);
+    p.background(0);
+
+    // Setting mode to degrees for spawning ringParticles in circle
+    p.angleMode(p.DEGREES);
+    // Sets 0,0 to centre
+    p.translate(p.width / 2, p.height / 2);
+    // Max number of ringParticles
+    p.frameRate(30);
+  };
+
+  p.windowResized = () => {
+    p.resizeCanvas(window.innerWidth, window.innerHeight, true);
+  };
+
+  p.draw = () => {
+    p.background(0, 0, 0, 25); // easy trails but deletes rings quickly
+
+    p.ellipse(p.mouseX,p.mouseY, 5,5);
+    if (p.random(1) < 0.03) {
+      fireworks.push(new Firework());
+    }
+
+    for (let i = fireworks.length - 1; i >= 0; i--) {
+      fireworks[i].update();
+      fireworks[i].show();
+
+      if (fireworks[i].done()) {
+        fireworks.splice(i, 1);
+      }
+    }
+
+    // If too many ringParticles on screen
+    if (ringParticles.length > MAX_RING_PARTICLES) {
+      // Delete old ringParticles (from beginning of array)
+      ringParticles.splice(0, p.floor((ringParticles.length - MAX_RING_PARTICLES * 0.8) * EASING));
+    }
+
+    // Loop through the array and show each particle
+    for (let i = 0; i < ringParticles.length; i++) {
+      ringParticles[i].update();
+      if (ringParticles[i].delete()) {
+        ringParticles.splice(i, 1);
+      } else {
+        ringParticles[i].show();
+      }
+    }
+  };
+  // Spawn ringParticles in ring with centre mouse x and y coords
+  p.mouseClicked = () => {
+    for (let i = 0; i < RING_COVERAGE; i += 4) {
+      const c = new RingParticle(p.mouseX, p.mouseY, i);
+      ringParticles.push(c);
+    }
+  };
+
+  class Firework {
+    public explodeSize: number;
+    public exploded: boolean;
+    private _firingParticle: Particle;
+    private _particles: Array<Particle>;
+
+    constructor() {
+      this.explodeSize = p.random(0, EXPLODE_RANGE);
+      this._firingParticle = new Particle(p.random(p.width), p.height, true);
+      this.exploded = false;
+      this._particles = [];
+    }
+
+    update() {
+      if (!this.exploded) {
+        this._firingParticle.update(gravity);
+
+        if (this._firingParticle.vel.y >= 0) {
+          this.explode();
+        }
+      }
+
+      for (let i = this._particles.length - 1; i >= 0; i--) {
+        this._particles[i].update(gravity);
+
+        if (this._particles[i].done()) {
+          this._particles.splice(i, 1);
+        }
+      }
```

Comment:
the following should work, but maybe double check if the return is right!

```suggestion
     this._particles = this._particles.filter((particle) => {
       particle.update(gravity);
       return !particle.done();
     }
```